File: Objective_C/Projects/Chp17_Memory Management and Automatic Reference Counting /MyView/MyView/ViewController.m
//
//  ViewController.m
//  MyView
//
//  Created by kongyunpeng on 12/21/15.
//  Copyright © 2015 kongyunpeng. All rights reserved.
//

#import "ViewController.h"
#import "MyView.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.data = [NSMutableArray array];
    _data = [[NSMutableArray array] retain];
    _data = [[NSMutableArray alloc] init];
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Swfit/Projects/MathOverload/MathOverload/ViewController.swift
//
//  ViewController.swift
//  MathOverload
//
//  Created by kongyunpeng on 12/26/15.
//  Copyright © 2015 kongyunpeng. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
       let maxtrix1 = Matrix2X2()
        maxtrix1.a11 = 3
        maxtrix1.a12 = 1
        maxtrix1.a21 = 1
        maxtrix1.a22 = 1
        let maxtrix2 = Matrix2X2()
        maxtrix2.a11 = 2
        maxtrix2.a12 = 2
        maxtrix2.a21 = 2
        maxtrix2.a22 = 2
     
        
     let matrix3 = maxtrix1 + maxtrix2
        print(matrix3.a11)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

